### PR TITLE
Reduce the size of the PrimitiveSet sent to the robots

### PR DESCRIPTION
**Pull request by nimazareian**
### Description
[Maximum transmission unit](https://en.wikipedia.org/wiki/Maximum_transmission_unit) (MTU) of ethernet is ~1500 bytes. With the headers that get added to a message, ~1400bytes is around the largest payload you can send in a single UDP packet, over the network. Currently, our PrimitiveSet packet sizes are around **1670-1700 bytes** long when we're commanding 6 robots with AI. This means that the network has to split our packet into 2 fragments and send them separately, then concatenate them back together once they both arrive at the destination (in this case, the Jetson on the robots). This fragmentation can increase the chance of packet loss, since if either of the fragments are lost or corrupted in the way, the entire packet will be dropped by the receiver. Additionally, the larger the size of the packet we send, the higher the risk of there being interference with one of the bytes, causing corruption of the message. To reduce the chance of overall packet loss, one approach we can take is to avoid [fragmentation](https://en.wikipedia.org/wiki/IP_fragmentation) by reducing our packet size.

Looking through the fields we send as part of the `PrimitiveSet` proto, it is easily noticeable that `static_obstacles` array takes the largest space. This field is used for drawing static obstacles (orange rectangles) in Thunderscope. Additionally, this field is not used by the robots since they dynamically recreate the set of static obstacles. As such, I have added a function to empty out the `static_obstacles` list before sending the packet over the network. With this update, the `PrimitiveSet` packet size has reduced significantly to around **720-780 bytes**. 

### Testing Done
Locally I tried to recreate sending the packet over wifi and it seem to still work, but someone should actually test this with actual robots to make sure that this does not mess up the primitive sets the AI sends.

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue

### CODE REVIEW

**Review by PhilD71 — APPROVED** [review]
Nice!

**Review by LiCody — COMMENTED** [review]
Could we add this to a robocup branch instead? dont think this should be permanently in the code. A cleaner solution would be converting motion constraints to polygon obstacle on the thunderscope side and removing the code that populates the static obstacles field on the ai side.

**Comment by PhilD71** [comment]
@LiCody My concern with leaving it in a robocup branch is that the changes cause master to diverge from robocup branch too much. Then we end up spending a lot of time re-integrating changes after the competition. I do agree that longterm we could find a better solution for this, but maybe creating an issue and tagging it with a #TODO might be an option? What do you think.

**Comment by LiCody** [comment]
> @LiCody My concern with leaving it in a robocup branch is that the changes cause master to diverge from robocup branch too much. Then we end up spending a lot of time re-integrating changes after the competition. I do agree that longterm we could find a better solution for this, but maybe creating an issue and tagging it with a #TODO might be an option? What do you think.

This change is very self contained so it should be easy to remove when merging the robocup branch to master. I am also fine with the second option

**Comment by nimazareian** [comment]
I think Cody brings up a fair point. This is not the cleanest solution. I do like the fact of removing the field completely, long term. If we notice that this change is helpful, I would probably opt for just having a second primitive type which is reduced sized (e.g. floats instead of doubles) and removes all fields not used by the robots... 
@itsarune It might be a good idea to start the Robocup branch soon, so we can start merging some of these changes in and we start having a unified branch for all field testing. We can pull changes that go into master as we go. What do you think?

**Comment by itsarune** [comment]
let's make that robocup branch on the day we start field testing (right now we're on pause till bordeaux-except for today).

File: src/software/thunderscope/robot_communication.py [hunk 1]
```diff
@@ -225,6 +227,21 @@ def __forward_to_proto_unix_io(self, type, data):
...
+        Reduces the size of the primitive by removing the static obstacles
+        :param primitive: the primitive to be reduced
+        :return: The reduced primitive
+        """
+
+        # The static_obstacles array is the largest part of the Primitive proto.
+        # Since it is not used by the robots, we will remove all values from its
+        # array.
+        if primitive.HasField("move"):
+            del primitive.move.motion_control.static_obstacles[:]
```
Comment (Bvasilchikov) [inline src/software/thunderscope/robot_communication.py:241]: From testing, this didn't seem to work (no diagnostics or play selected). Maybe try setting the static obstacles to an empty array?

**Comment by itsarune** [comment]
this is pretty hacky, so yeah let's put this in the RoboCup branch

File: src/software/thunderscope/robot_communication.py (hunk 1 above)
Comment (nimazareian) [reply src/software/thunderscope/robot_communication.py:241]: This is different than the initial field testing you did with the patch I sent. Have you tested this updated version as well?

**Comment by nimazareian** [comment]
Closing this for now in favour of a new and smaller msg type to the robots.

**Pull request closed without merging**